### PR TITLE
Support virtual study in new study-view

### PR DESCRIPTION
**Pull request by kalletlak**
Fix : https://github.com/cBioPortal/cbioportal/issues/4463

- [x] Support visualizing virtual study or any combination
- [x] Show error message when samples in virtual study are no longer available in the physical studies
- [x] Virtual Study button tooltip

### CODE REVIEW

File: src/pages/studyView/StudyViewPage.tsx [hunk 1]
```diff
@@ -172,6 +175,18 @@ export default class StudyViewPage extends React.Component<IStudyViewPageProps,
                              className="mainTabs">
 
                         <MSKTab key={0} id="summaryTab" linkText="Summary">
+                            {
+                                this.store.invalidSampleIds.length > 0 &&
+                                this.showErrorMessage &&
+                                <div>
+                                    <div className="alert alert-danger">
+                                        <button type="button" className="close" onClick={(event) => this.showErrorMessage = false}>&times;</button>
+                                        Following sample(s) might have been deleted/updated with the recent data updates
```
Comment (zhx828) [inline src/pages/studyView/StudyViewPage.tsx]: I think the language can be easily updated base on the invalidSampleIds length.

File: src/pages/studyView/StudyViewPageStore.ts [hunk 2]
```diff
@@ -286,25 +287,161 @@ export class StudyViewPageStore {
...
+
+    readonly allPhysicalStudies = remoteData({
+        invoke: async () => {
+            return await defaultClient.getAllStudiesUsingGET({ projection: 'SUMMARY' });
+        },
+        default: []
+    });
+
+    @computed get physicalStudiesSet() {
+        return _.reduce(this.allPhysicalStudies.result, (acc, next) => {
```
Comment (zhx828) [inline src/pages/studyView/StudyViewPageStore.ts]: why not just _.keyBy?

File: src/pages/studyView/StudyViewPageStore.ts [hunk 3]
```diff
@@ -286,25 +287,161 @@ export class StudyViewPageStore {
...
+                name: virtualStudy.data.name,
+                description: virtualStudy.data.description,
+                studyId: virtualStudy.id,
+            } as CancerStudy;
+        });
+
+        return [...this.filteredPhysicalStudies, ...filteredVirtualStudies];
+    }
+
+    @computed get unKnownQueriedIds() {
```
Comment (zhx828) [inline src/pages/studyView/StudyViewPageStore.ts]: unknownQueriedIds

File: src/pages/studyView/StudyViewPageStore.ts [hunk 4]
```diff
@@ -321,9 +458,9 @@ export class StudyViewPageStore {
     }
 
     readonly clinicalAttributes = remoteData({
-        await: () => [this.studies],
+        await: () => [this.filteredVirtualStudies],
         invoke: () => defaultClient.fetchClinicalAttributesUsingPOST({
-            studyIds: this.studies.result.map(study => study.studyId)
+            studyIds: this.queriedPhysicalStudies
```
Comment (zhx828) [inline src/pages/studyView/StudyViewPageStore.ts]: Why we wait for filteredVirtualStudies when the API call is only on filteredVirtualStudies?

File: src/pages/studyView/StudyViewPageStore.ts [hunk 5]
```diff
@@ -286,25 +287,161 @@ export class StudyViewPageStore {
...
+                        sampleId,
+                        studyId
+                    }
+                })
+            ));
         },
         default: []
     });
 
+    @computed get emptyFilter() {
```
Comment (zhx828) [inline src/pages/studyView/StudyViewPageStore.ts]: Why don't we call it `studyViewFilter`?

File: src/pages/studyView/StudyViewPageStore.ts [hunk 6]
```diff
@@ -286,25 +287,161 @@ export class StudyViewPageStore {
...
+                    studyIds: this.queriedPhysicalStudies
                 } as MolecularProfileFilter
             })
         },
         default: []
     });
 
-    readonly studies = remoteData({
+
+    readonly allPhysicalStudies = remoteData({
```
[the remark this answers is not in the document]
Comment (alisman) [reply src/pages/studyView/StudyViewPageStore.ts]: it's true that we will be loading studies on various pages.  We'll need some central store to deal with this.  But we will also be caching these at client level, so it's not so important.

File: src/pages/studyView/StudyViewPageStore.ts (hunk 2 above)
Comment (kalletlak) [reply src/pages/studyView/StudyViewPageStore.ts]: 👍

File: src/pages/studyView/StudyViewPage.tsx (hunk 1 above)
Comment (alisman) [reply src/pages/studyView/StudyViewPage.tsx]: Also add "The ..."

**Pull request merged**